### PR TITLE
Beacon tower change

### PR DESCRIPTION
**Pull request by Jagnt**

### CODE REVIEW

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 1]
```diff
@@ -99,6 +99,18 @@ function End(ply)
     ply:PrintMessage(HUD_PRINTTALK, "Stopping game...")
 end
 
+function NearBeacon(ply)
+    local beacons = ents.FindByClass("horde_watchtower_beacon")
+    if beacons and #beacons > 0 then
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_commands.lua:104]: ents.FindByClass always returns a table, it'd be better to early return here. You'd have to fix the `end` and scoping once you accept this
```suggestion
    if #beacons == 0 then return false end
```

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 2]
```diff
@@ -99,6 +99,18 @@ function End(ply)
     ply:PrintMessage(HUD_PRINTTALK, "Stopping game...")
 end
 
+function NearBeacon(ply)
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_commands.lua:102]: this should not be a global function
```suggestion
local function nearBeacon(ply)
```

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 3]
```diff
@@ -107,7 +119,7 @@ function Shop(ply)
     if ply:Alive() then
         local res = hook.Run("Horde_OnPlayerOpenShop", ply)
         if res ~= true then
-            if HORDE.current_break_time <= 0 then
+            if HORDE.current_break_time <= 0 and not NearBeacon(ply) then
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_commands.lua:122]:
```suggestion
            if HORDE.current_break_time <= 0 and not nearBeacon(ply) then
```

Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua:121]: all the other functions here used the same case

File: gamemodes/horde/gamemode/sv_commands.lua (hunk 1 above)
Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua]: added this

Comment (wrefgtzweve) [reply gamemodes/horde/gamemode/sv_commands.lua]: you still check if beacons exists

**Review by wrefgtzweve — APPROVED** [review]

**Pull request merged**